src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -603,8 +603,21 @@ def _report_single_requirement_conflict(
 
         cands = self._finder.find_all_candidates(req.project_name)
         skipped_by_requires_python = self._finder.requires_python_skipped_reasons()
-        versions = [str(v) for v in sorted({c.version for c in cands})]
 
+        is_yanked: Dict[bool, Set[CandidateVersion]] = {True: set(), False: set()}
```

Comment:
This dict feels a little bit awkward to me. I'd have done sets and a good old if. It might be a matter of personal preference though.

Reply:
agreed, removed 👍